### PR TITLE
test: integration test for tekton on cluster build

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :broom: Integration test for tekton on cluster build.

### CODE REVIEW

File: hack/allocate.sh [hunk 1]
```diff
@@ -71,6 +71,8 @@ containerdConfigPatches:
 - |-
   [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:50000"]
```
Comment (matejvasek) [inline hack/allocate.sh:72]: @jrangelramos changes here might be of interest for you. We might get rid of `ttl.sh` in future.

**Comment by matejvasek** [comment]
Great s2i on cluster build just finished with:
```
        error preparing container for next step: error creating build container: error creating container: creating read-write layer with ID "d531de0ae4a8d9be6d6b28a755cd99073ac26dcfc9b85cd4110df14c6059e9df": no space left on device
```

**Comment by matejvasek** [comment]
@zroubalik @jrangelramos PTAL

**Comment by matejvasek** [comment]
> Great, `s2i` on cluster build just crashed with:
> 
> ```
>         error preparing container for next step: error creating build container: error creating container: creating read-write layer with ID "d531de0ae4a8d9be6d6b28a755cd99073ac26dcfc9b85cd4110df14c6059e9df": no space left on device
> ```

It happened at least 3 times in row, not good.
How does is come that e2e with s2i works? @jrangelramos

**Review by zroubalik — APPROVED** [review]
/lgtm

File: pkg/pipelines/resources/tekton/task/func-s2i/0.1/func-s2i.yaml
[the remark this answers is not in the document]
Comment (lance) [reply pkg/pipelines/resources/tekton/task/func-s2i/0.1/func-s2i.yaml:1]: I have been wondering about this as well. I think it is not a bad idea.

**Comment by matejvasek** [comment]
> > > Great, `s2i` on cluster build just crashed with:
> > > ```
> > >         error preparing container for next step: error creating build container: error creating container: creating read-write layer with ID "d531de0ae4a8d9be6d6b28a755cd99073ac26dcfc9b85cd4110df14c6059e9df": no space left on device
> > > ```
> > 
> > 
> > It happened at least 3 times in row, not good. How does is come that e2e with s2i works? @jrangelramos
> 
> Hmm.. how big gitlab is.. ? I remember trying that a long ago and it used to be big. Might be consuming host running space..

@jrangelramos  It takes 1-2 GB of memory (RAM) not sure about disk.

**Comment by jrangelramos** [comment]
I'm not sure we can have a gh runner with larger disk space.. Perhaps you could stick your tests with `pack` builder for now. s2i task is also tested against all supported runtime e2e on cluster test and runtime tests.. so we should be covered.

**Comment by matejvasek** [comment]
@lance @jrangelramos for now I decided to skip s2i test.
Please re-lgtm and override the linter error.

**Comment by jrangelramos** [comment]
@matejvasek what about increasing func-s2i to 0.2? we could start incrementing these yaml version whenever we modify them.. wdty?

**Comment by matejvasek** [comment]
> > @matejvasek what about increasing func-s2i to 0.2? we could start incrementing these yaml version whenever we modify them.. wdty?
> 
> @jrangelramos TBH I am not even sure we want to have the version on file path. So I would prefer leave it and maybe discus later.

@jrangelramos look for instance on https://github.com/openshift-pipelines/pipelines-as-code , they have yamls in `./config` and on release cut they compile uber-yaml that is then available on URL containing version/branch/tag.

IMO having version as named directory would make sense if we were maintaining multiple version of task at once (not just rename on each update).

So I would prefer to remove version from file path entirely.

**Comment by matejvasek** [comment]
@zroubalik or @lance  please override the linter error.

File: pkg/pipelines/resources/tekton/task/func-s2i/0.1/func-s2i.yaml
[the remark this answers is not in the document]
Comment (matejvasek) [reply pkg/pipelines/resources/tekton/task/func-s2i/0.1/func-s2i.yaml:1]: I am not sure about this:
https://github.com/knative/func/pull/1789#issuecomment-1581428377
https://github.com/knative/func/pull/1789#issuecomment-1581434100

**Pull request merged**